src/pyflex/window_selector.py

```diff
@@ -911,12 +992,36 @@ def plot(self, filename=None):
         ax.spines['right'].set_color('none')
         ax.spines['left'].set_color('none')
         ax.spines['top'].set_color('none')
+        stepsize = 200
+        start, end = ax.get_xlim()
+        ax.xaxis.set_ticks(np.arange(start, end+1, stepsize))
         ax.set_yticks([])
         ax.xaxis.set_ticks_position('bottom')
 
         plt.text(0.01, 0.99, 'STA/LTA', horizontalalignment='left',
                  verticalalignment='top', transform=ax.transAxes)
 
+        # print more information on the figure
+        text = "Station: %s.%s     Comp: %s" % (self.observed.stats.network, self.observed.stats.station,
+            self.observed.stats.channel)
+        plt.text(0.75, 0.99, text, horizontalalignment='left',
+                 verticalalignment='top', transform=ax.transAxes)
+        dist_in_m, az, baz = geodetics.gps2DistAzimuth(self.event.latitude, self.event.longitude,
+                                self.station.latitude, self.station.longitude)
+        dist_in_degree = dist_in_m / (1000.0 * 40075.) * 360.0
+        text = "Dist: %5.2f$^\circ$   Azimuth: %5.2f$^\circ$" % ( dist_in_degree, az)
+        plt.text(0.75, 0.90, text, horizontalalignment='left',
+                 verticalalignment='top', transform=ax.transAxes)
+        offset = self.event.origin_time - self.observed.stats.starttime
+        signal_start = self.config.signal_start_index * self.observed.stats.delta + offset
+        signal_end = self.config.signal_end_index * self.observed.stats.delta + offset
+        #print(self.config.noise_end_index, self.config.signal_start_index)
+        #print(self.config.signal_end_index, self.config.signal_start_index)
+        #print(signal_start, signal_end)
+        text = "Signal Zone(s): [%-6.1f, %6.1f]" % (signal_start, signal_end)
+        plt.text(0.75, 0.81, text, horizontalalignment='left',
+                 verticalalignment='top', transform=ax.transAxes)
```

Comment:
Additionally it would also be great to see the time limits imposed by the various travel time limitations. `all_waves`, `body_waves`, ... then it is visually simpler to judge what is going on.